src/include/global_kvs.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Filename: global_kvs.h
+ * Description: Declaration for NSAL global kvs api
+ *
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ */
+ 
+/** Global KV-Store operations **/
+/* The functionality provided by below APIs is dependent on basic
+ * framework provided kvstore.h
+ */ 
+
+#ifndef GLOBAL_KVS_H
+#define GLOBAL_KVS_H
+
+
+/* Set an index mentioned by fid_str as global kvstore*/
+int nsal_set_global_kvs(const char *fid_str);
+
+/* Allocate memory */
+int nsal_global_alloc(void **ptr, size_t size);
+
+/* Free the allocated memory */
+void nsal_global_free(void *ptr);
+
+/* Store KV pair in the global kvs */
+int nsal_global_set_kv(void *key, const size_t klen,
+		       void *value, const size_t vlen);
```

Comment:
Suggestion: Here instead of passing separate key, klen, value, vlen, can we make use of the following structures?

https://github.com/Seagate/cortx-nsal/blob/e4af34e9f750f533b6e43e55f1d68da1caa65692/src/include/kvstore.h#L174
this kvpair structure holds both key and value along with its length. It also includes helper APIs kor kvpair like alloc, init, free. 

OR

https://github.com/Seagate/cortx-utils/blob/3abf9ea5b776a5a31bd35387ff8ea6348706897b/c-utils/src/include/utils.h#L29
This structure is a simple one to use and can hold data and it's length together. Its helper function is buff_init().
here's the use case of this structure :- https://github.com/Seagate/cortx-nsal/blob/e4af34e9f750f533b6e43e55f1d68da1caa65692/src/include/kvnode.h#L218

Reply:
Is it fine, if we treat this as an enhancement and address it via another ticket ?

Reply:
Sure, that is okay for me.